### PR TITLE
Modify rule S1117: Update to LaYC

### PR DESCRIPTION
**Pull request by cristian-ambrosini-sonarsource**
## Review

A dedicated reviewer checked the rule description successfully for:

- [ ] logical errors and incorrect information
- [ ] information gaps and missing content
- [ ] text style and tone
- [ ] PR summary and labels follow [the guidelines](https://github.com/SonarSource/rspec/#to-modify-an-existing-rule)

### CODE REVIEW

File: rules/S1117/csharp/metadata.json [hunk 1]
```diff
@@ -1,3 +1,4 @@
 {
-  "title": "Local variables should not shadow class fields"
```
Comment (gregory-paidis-sonarsource) [inline rules/S1117/csharp/metadata.json:2]: Shouldn't this be `class fields or properties` ?

File: rules/S1117/csharp/rule.adoc [hunk 2]
```diff
@@ -1,6 +1,8 @@
+:message: Rename "${elementName}" which hides the (field|property) with the same name.
+
 == Why is this an issue?
 
-include::../description.adoc[]
+Overriding or shadowing a field or a property declared in an outer scope can strongly impact the readability, and therefore the maintainability, of a piece of code. Developers may mistakenly assume they are modifying or accessing the class field/property when, in fact, they are working with the local variable.
 
 === Noncompliant code example
```
Comment (gregory-paidis-sonarsource) [inline rules/S1117/csharp/rule.adoc:7]: You forgot `How to fix it` and `Code examples` section headers

**Review by gregory-paidis-sonarsource — CHANGES_REQUESTED** [review]
Couple of small changes needed

File: rules/S1117/csharp/rule.adoc (hunk 2 above)
Comment (cristian-ambrosini-sonarsource) [reply rules/S1117/csharp/rule.adoc:7]: I forgot to remove `=== Noncompliant code example title`. I don't think those sections are needed in this particular rule (the fix is obvious and I'm not providing a compliant snippet)

File: rules/S1117/csharp/metadata.json (hunk 1 above)
Comment (cristian-ambrosini-sonarsource) [reply rules/S1117/csharp/metadata.json:2]: Yeah it should; I'm changing it here in the metadata, should I change it in some other place as well?

**Review by gregory-paidis-sonarsource — COMMENTED** [review]
LGTM

**Pull request merged**